src/item.h

```diff
@@ -1905,33 +1906,23 @@ class item : public visitable<item>
         void set_birthday( const time_point &bday );
         void handle_pickup_ownership( Character &c );
         int get_gun_ups_drain() const;
-        inline void set_old_owner( const faction *temp_owner ) {
+        inline void set_old_owner( const faction_id temp_owner ) {
```

Comment:
```suggestion
        inline void set_old_owner( const player &temp_owner ) {
```
and then `old_owner = temp_owner.get_faction().id();`

Reply:
Actually, I guess you need both versions, but alternate arguments are cheap.